R/DBConnection.R

```diff
@@ -106,6 +106,103 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' Execute a non-select query
+#'
+#' This function should be used when you want to execute a non-
+#' \code{SELECT} query on table (ex: \code{UPDATE}, \code{DELETE},
+#' \code{INSERT INTO}, \code{DROP TABLE}, ...). It will execute
+#' the query and return the number of rows affected by the operation.
+#'
+#' @inheritParams dbDisconnect
+#' @param statement a character vector of length 1 containing SQL.
+#' @return The number of rows affected by the \code{statement}
+#' @aliases dbExecQuery,DBIConnection,character-method
+#' @family connection methods
+#' @export
+#' @examples
+#' con <- dbConnect(RSQLite::SQLite(), ":memory:")
+#' dbWriteTable(con, "cars", head(cars, 3))
+#' dbReadTable(con, "cars")   # there's 3 rows!
+#' dbExecQuery(con, "INSERT INTO cars (speed, dist)
+#'                   VALUES (1, 1), (2, 2), (3, 3);")
+#' dbReadTable(con, "cars")   # there's now 6 rows!
+#' dbDisconnect(con)
+setGeneric("dbExecQuery",
+  def = function(conn, statement, ...) standardGeneric("dbExecQuery")
+)
+
+#' @export
+setMethod("dbExecQuery", signature("DBIConnection", "character"),
+  function(conn, statement, ...) {
+    if (grepl("select", tolower(statement))) {
```

Comment:
Should we not validate at all? And accept everything? In that case, even if the "query" is reading data, since we don't use `dbFetch` (and subsequently also do not return the resulting dataframe), there's no harm done...(?) Except that shouldn't we be a bit more assertive about this? Maybe if your "query" does not affect any rows (`dbGetRowsAffected(res) == 0`), then throw an error (or warning...)?